### PR TITLE
Changed Location as Area

### PR DESCRIPTION
**Pull request by yashodgayashan**

### CODE REVIEW

**Comment by deshankoswatte** [comment]
@yashodgayashan now do we get the area generated onto the schema by the openpi gen-client or should we add that manually?

**Comment by yashodgayashan** [comment]
> @yashodgayashan now do we get the area generated onto the schema by the openpi gen-client or should we add that manually?

Forget to do it. I will correct it and commit

**Comment by sanjulamadurapperuma** [comment]
> > @yashodgayashan now do we get the area generated onto the schema by the openpi gen-client or should we add that manually?
> 
> Forget to do it. I will correct it and commit

Yeah other than this everything else looks good to me

File: core/src/env_platform/env_service.bal [hunk 1]
```diff
@@ -1,11 +1,12 @@
...
-import ballerina/openapi;
+// import ballerina/openapi;
+// import ballerina/io;
 
 listener http:Listener ep0 = new (9090);
 
-@openapi:ServiceInfo {
-    contract: "resources/openapi_v3.yaml"
-}
+// @openapi:ServiceInfo {
```
Comment (sanjulamadurapperuma) [inline core/src/env_platform/env_service.bal]: Any reason why these were commented out? If they are not needed anymore we can remove them.

**Comment by yashodgayashan** [comment]
Resolved requested changes @deshankoswatte @sanjulamadurapperuma

File: core/src/env_platform/resources/environment_platform.postman_collection.json [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+					"path": [
+						"applications"
+					]
+				}
+			},
+			"response": []
+		}
+	],
+	"protocolProfileBehavior": {}
+}
```
Comment (sanjulamadurapperuma) [inline core/src/env_platform/resources/environment_platform.postman_collection.json:41]:
```suggestion
}

```

File: core/src/env_platform/env_service.bal [hunk 3]
```diff
@@ -117,7 +118,7 @@ service envservice on ep0 {
         path: "/maps/validate-map",
         body: "body"
     }
-    resource function validateCoordinates(http:Caller caller, http:Request req, Location body) returns error? {
+    resource function validateTheLocation(http:Caller caller, http:Request req, Location[] body) returns error? {
```
Comment (deshankoswatte) [inline core/src/env_platform/env_service.bal:121]: Here it should be `Area` instead of `Location` right?

Comment (sanjulamadurapperuma) [inline core/src/env_platform/env_service.bal:121]:
```suggestion
    resource function validateTheArea(http:Caller caller, http:Request req, Location[] body) returns error? {
```

Comment (sanjulamadurapperuma) [reply core/src/env_platform/env_service.bal:121]: Oh just saw this, yeah it should be right?

File: core/src/env_platform/env_service.bal (hunk 1 above)
Comment (yashodgayashan) [reply core/src/env_platform/env_service.bal]: Could you please check with the latest commits I have resolved it.

File: core/src/env_platform/env_service.bal (hunk 3 above)
Comment (deshankoswatte) [reply core/src/env_platform/env_service.bal]: @yashodgayashan I'm confused cause we have both `Area` and `Location` can you explain the use of these two to us?

File: core/src/env_platform/schema.bal [hunk 4]
```diff
@@ -16,7 +16,7 @@ type TreeRemovalApplication record {
     string district;
     string nameOfTheLand;
     string planNumber;
-    Location[] location;
+    Location[] area;
```
Comment (sanjulamadurapperuma) [inline core/src/env_platform/schema.bal:19]:
```suggestion
    Area[] area;
```
Shouldn't this be made Area?
It's there in the rest of the code as well

File: core/src/env_platform/env_service.bal (hunk 3 above)
Comment (yashodgayashan) [reply core/src/env_platform/env_service.bal]: Location is a point where it has latitude and longitude. Area is the set of locations which will cover an area.

Comment (deshankoswatte) [reply core/src/env_platform/env_service.bal]: So does that mean the location includes the area?

Comment (sanjulamadurapperuma) [reply core/src/env_platform/env_service.bal]: Area is a collection of Location objects. It's used to construct the polygon. Correct me if I am wrong @yashodgayashan

Comment (yashodgayashan) [reply core/src/env_platform/env_service.bal]: I didn't get it. Simply I mean Area is an array of locations.

Comment (yashodgayashan) [reply core/src/env_platform/env_service.bal]: > Area is a collection of Location objects. It's used to construct the polygon. Correct me if I am wrong @yashodgayashan

yes it is.

**Pull request merged**